data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/BinlogEventListener.java

```diff
@@ -82,15 +89,16 @@ public BinlogEventListener(final Buffer<Record<Event>> buffer,
                                final BinaryLogClient binaryLogClient,
                                final StreamCheckpointer streamCheckpointer,
                                final AcknowledgementSetManager acknowledgementSetManager) {
+        this.buffer = buffer;
         this.binaryLogClient = binaryLogClient;
         tableMetadataMap = new HashMap<>();
         recordConverter = new StreamRecordConverter(sourceConfig.getStream().getPartitionCount());
-        bufferAccumulator = BufferAccumulator.create(buffer, DEFAULT_BUFFER_BATCH_SIZE, BUFFER_TIMEOUT);
         s3Prefix = sourceConfig.getS3Prefix();
         tableNames = sourceConfig.getTableNames();
         isAcknowledgmentsEnabled = sourceConfig.isAcknowledgmentsEnabled();
         this.pluginMetrics = pluginMetrics;
         pipelineEvents = new ArrayList<>();
+        executorService = Executors.newCachedThreadPool();
```

Comment:
I think we want to have an upper bound on this. And make it configurable.

Please use `BackgroundThreadFactory` to make tracking threads easier and promote shutdown.

Example:

https://github.com/opensearch-project/data-prepper/blob/afe84648d662ddc469a6cbf9f22ccd6ff9dfda2c/data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/SqsService.java#L52

Reply:
Done.